itest/loadtest/load_test.go

```diff
@@ -5,10 +5,28 @@ package loadtest
 import (
 	"context"
 	"testing"
+	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
 	"github.com/stretchr/testify/require"
 )
 
+var (
+	testDuration = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
```

Comment:
We should make this a histogram metric. Then we'll be able to do percentile plots, and heat maps, etc.

Reply:
So I looked into this and a histogram won't persist over multiple runs (it will be overwritten across runs). There's some tricks we can try to get the gateway to persist but really not worth the time & diff.

Given the frequency at which we run the loadtests, we can produce a histogram from the GaugeVec with PromQL directly in Grafana. This is not as performant as directly using a histogram, but will give us the same insights on percentiles etc

Reply:
Are you referring to [this](https://github.com/prometheus/pushgateway#:~:text=For%20the%20most,of%20the%20Pushgateway).)? 

IIUC that means that if the pushgateway is restart before it's scraped, the metrics won't persist, but IIUC we have the system always running.

Reply:
yeah in our setup pushgateway should always be alive. I was referring to the part where the client side restarts (by design) and we push a fresh instance of "histogram". See [here](https://github.com/prometheus/pushgateway/issues/65)

By pushing a fresh histogram to the pushgateway we're overwriting the old one, effectively only keeping the values of our last run (last-write-wins)

I'm not sure if a HistogramVec, with unique label per test run, is going to be a fine workaround, less performant than a simple histogram nevertheless

Reply:
There's also a community implementation of pushgateway that seems to be solving this issue (haven't tested it)
https://github.com/zapier/prom-aggregation-gateway